scen_edit/gadget.lua

```diff
@@ -121,7 +121,16 @@ function gadget:Initialize()
 
     -- detect game mode
     local modOpts = Spring.GetModOptions()
-    local sb_gameMode = (tonumber(modOpts.sb_gameMode) or 0)
+    if modOpts.sb_gameMode == nil and modOpts.play_mode ~= nil then
+        -- That's not actually stright-forward, but so far a smart alternative.
```

Comment:
I'd say this long comment block is unnecessary (although I get now what you're trying to fix).
I think it would be simply enough to state:
-- Report outdated script setup with missing sb_gameMode value